tests/js/client/shell/shell-dropped-followers-elcheapo-commit-cluster.js

```diff
@@ -0,0 +1,339 @@
+      getEndpointsByType,
+      debugCanUseFailAt,
+      debugSetFailAt,
+      debugClearFailAt
+    } = require('@arangodb/test-helper');
+
+function createCollectionWithTwoShardsSameLeaderAndFollower(cn) {
+  db._create(cn, {numberOfShards:2, replicationFactor:2});
+  // Get dbserver names first:
+  let health = arango.GET("/_admin/cluster/health").Health;
+  let endpointMap = {};
+  for (let sid in health) {
+    endpointMap[health[sid].ShortName] = health[sid].Endpoint;
+  }
+  let plan = arango.GET("/_admin/cluster/shardDistribution").results[cn].Plan;
+  let shards = Object.keys(plan);
+  let coordinator = "Coordinator0001";
+  let leader = plan[shards[0]].leader;
+  let follower = plan[shards[0]].followers[0];
+  // Make leaders the same:
+  if (leader !== plan[shards[1]].leader) {
+    let moveShardJob = {
+      database: "_system",
+      collection: cn,
+      shard: shards[1],
+      fromServer: plan[shards[1]].leader,
+      toServer: leader,
+      isLeader: true
+    };
+    let res = arango.POST("/_admin/cluster/moveShard", moveShardJob);
+    let start = internal.time();
+    while (true) {
+      if (internal.time() - start > 120) {
+        assertTrue(false, "timeout waiting for shards being in sync");
+        return;
+      }
+      let res2 = arango.GET(`/_admin/cluster/queryAgencyJob?id=${res.id}`);
+      if (res2.status === "Finished") {
+        break;
+      }
+      internal.wait(1);
+    }
+  }
+  plan = arango.GET("/_admin/cluster/shardDistribution").results[cn].Plan;
+  // Make followers the same:
+  if (follower !== plan[shards[1]].followers[0]) {
+    let moveShardJob = {
+      database: "_system",
+      collection: cn,
+      shard: shards[1],
+      fromServer: plan[shards[1]].followers[0],
+      toServer: follower,
+      isLeader: false
+    };
+    let res = arango.POST("/_admin/cluster/moveShard", moveShardJob);
+    let start = internal.time();
+    while (true) {
+      if (internal.time() - start > 120) {
+        assertTrue(false, "timeout waiting for shards being in sync");
+        return;
+      }
+      let res2 = arango.GET(`/_admin/cluster/queryAgencyJob?id=${res.id}`);
+      if (res2.status === "Finished") {
+        break;
+      }
+      internal.wait(1);
+    }
+  }
+  return { endpointMap, coordinator, leader, follower, shards };
+}
+
+function waitForShardsInSyncAgain(cn) {
+  let start = internal.time();
+  while (true) {
+    if (internal.time() - start > 60) {
+      assertTrue(false, "timeout waiting for shards being in sync");
+      return;
+    }
+    let sd = arango.GET("/_admin/cluster/shardDistribution");
+    sd = sd.results[cn];
+    let plan = sd.Plan;
+    let current = sd.Current;
+    let shards = Object.keys(plan);
+    let good = 0;
+    for (let s of shards) {
+      if (plan[s].followers.length === current[s].followers.length) {
+        ++good;
+      }
+    }
+    if (good === shards.length) {
+      break;
+    }
+    internal.wait(1);
+  }
+}
```

Comment:
This is essentially the same as `waitForShardsInSync` in the other test, can we move them into `test-helper.js` or so to deduplicate?